service_crategen/src/commands/generate/codegen/rest_json.rs

```diff
@@ -119,13 +102,14 @@ impl GenerateProtocol for RestJsonGenerator {
         writeln!(writer, "use serde_json;")?;
 
         // avoid unused imports when building services that don't use params
-        if service.service_type_name() != "Batch" {
+        if service_has_query_parameters(service) {
```

Comment:
An alternative here that wouldn't require all the query param checking is to just not import these and instead using the fully qualified path (i.e. `rusoto_core::param::Param`) in the generated code. Also has the advantage of ensuring there will never be a naming collision with a potential `Params` or `ServiceParams` type in the future.